File: examples/python/basic_test.py
import requests
import simplejson
import os
from time import sleep
from datetime import datetime, timedelta
from dateutil.tz import tzlocal

API_URL = "http://127.0.0.1:8000/api/v1/job/"

data = {
    "name": "test_job",
    "command": "bash " + os.path.dirname(os.path.realpath(__file__)) + "/../example-kala-commands/example-command.sh",
    "epsilon": "PT5S",
}

dt = datetime.isoformat(datetime.now(tzlocal()) + timedelta(0, 10))
data["schedule"] = "%s/%s/%s" % ("R2", dt, "PT10S")

if __name__ == "__main__":
    print('Sending request to {0}'.format(API_URL))
    print('Payload is: {0}'.format(data))
    r = requests.post(API_URL, data=simplejson.dumps(data))

    print("\n\nCreating...\n")
    job_id = simplejson.loads(r.content)['id']
    print('Job was created with an id of {0}'.format(job_id))

    print("\n\nGetting...\n")
    m = requests.get(API_URL + job_id)
    print(m.content)

    print("\n\nDeleting...\n")

    print("\n\nWaiting to delete...\n")
    sleep(21)
    n = requests.delete(API_URL + job_id)

    print(n.__dict__)

File: examples/ruby/basic_test.rb
require 'net/http'
require 'time'
require 'json'

class Kala
  API_URL = 'http://127.0.0.1:8000/api/v1/job/'

  def post(data)
    req = Net::HTTP::Post.new(uri.path)
    req.body = data.to_json
    call(req)
  end

  def get(id)
    req = Net::HTTP::Get.new("#{uri.path}#{id}/")
    call(req)
  end

  def delete(id)
      req = Net::HTTP::Delete.new("#{uri.path}#{id}/")
    call(req)
  end

  private

  def uri
    @uri ||= URI(API_URL)
  end

  def call(req)
    res = Net::HTTP.start(uri.host, uri.port) do |http|
      http.request(req)
    end

    if res.body
      JSON.parse(res.body)
    else
      res
    end
  end
end

data = {
  name: 'ruby_job',
  command: "bash #{Dir.pwd}/../example-kala-commands/example-command.sh",
  epsilon: 'PT5S',
  schedule: "R2/#{(Time.now + 10).iso8601}/PT10S"
}

puts "Sending request to #{Kala::API_URL}"
puts "Payload is: #{data}\n\n"

kala = Kala.new

job_id = kala.post(data)['id']
puts "Job was created with an id of #{job_id}"

puts "Getting informations about job #{job_id}"
puts kala.get(job_id)

puts "Waiting to delete job #{job_id}"
sleep(21)
puts kala.delete(job_id).inspect
